starboard/shared/starboard/player/video_dmp_reader.cc

```diff
@@ -122,12 +119,6 @@ VideoDmpReader::VideoDmpReader(
     video_access_units_.reserve(dmp_info_.video_access_units_size);
     return;
   }
-
-  Parse();
-
-  if (!already_cached) {
-    GetRegistry()->Register(file_reader_.GetAbsolutePathName(), dmp_info_);
```

Comment:
Maybe we can rename Register() to TryRegister() and only update the reference when there isn't one?  Since the current implementation of Register() is already under a lock, this change can be quite straight-forward.

Another way is to not use a lock in Registry, and move the lock into VideoDmpReader::VideoDmpReader and name it registry_lock_.